experiments/kompanion/cmd/export/export.go

```diff
@@ -141,66 +248,60 @@ func run(_ *cobra.Command, _ []string) {
 	sort.Slice(resources, func(i, j int) bool {
 		return resources[i].SingularName < resources[j].SingularName
 	})
+
+	// todo acpana debug logs
 	// log.Printf("Going to iterate over the following resources %+v", resourcesToName(resources))
 
 	namespaces, err := clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		log.Fatalf("Error fetching namespaces: %s\n", err)
 	}
 
-	reportName := timestampedName("report")
+	reportName = timestampedName("report")
 	if err := os.Mkdir(filepath.Join(reportName), os.ModePerm); err != nil {
 		log.Fatalf("Could not create \"report\" directory: %w", err)
 	}
+
+	// create the work log for go routine workers to use
+	workLog = &namespacesWorkLog{}
 	for _, ns := range namespaces.Items {
+		workLog.namespaces = append(workLog.namespaces, ns.Name)
+	}
 
-		log.Printf("Processing namespace: %s", ns.Name)
-		if shouldExclude(ns.Name, ignoreNamespaces, targetNamespaces) {
-			continue
+	// todo acpana parametrize as a flag
+	log.Printf("Starting worker threads to process namespaces")
+	var wg sync.WaitGroup
+	for i := 0; i < workerRountines; i++ {
+		if workLog.Done() {
+			// if there's no more work stop spawning worker routines;
+			// we're done.
+			break
 		}
 
-		if err := os.Mkdir(filepath.Join(reportName, ns.Name), os.ModePerm); err != nil {
-			log.Printf("Error creating directory for namespace %s:, err: %w; skipping", ns.Name, err)
-			continue
-		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
 
-		for _, res := range resources {
-			gvr := schema.GroupVersionResource{
-				Group:    res.Group,
-				Version:  res.Version,
-				Resource: res.Name,
+			if workLog.Done() {
+				return
 			}
 
-			resources, err := dynamicClient.Resource(gvr).Namespace(ns.Name).List(context.TODO(), metav1.ListOptions{})
-			if err != nil {
-				if apierrors.IsNotFound(err) {
-					log.Printf("gvr %s not found in namespace %s", gvr, ns.Name)
-					continue
-				} else {
-					log.Printf("Error fetching gvr %s resources in namespace %s: %s\n", gvr, ns.Name, err)
-					continue
-				}
-			}
-
-			for _, r := range resources.Items {
-				if shouldExclude(r.GetName(), ignoreObjects, targetObjects) {
-					continue
+			for {
+				ns := workLog.GetWork()
+				if ns == "" {
+					// no work
+					return
 				}
 
-				filename := filepath.Join(reportName, ns.Name, r.GetName()+".yaml")
-				data, err := yaml.Marshal(r)
-				if err != nil {
-					log.Printf("Error marshalling resource %s in namespace %s: %s\n", r.GetName(), ns.Name, err)
-					continue
-				}
-				err = ioutil.WriteFile(filename, data, 0o644)
-				if err != nil {
-					log.Printf("Error writing file for resource %s in namespace %s: %s\n", r.GetName(), ns.Name, err)
-				}
+				processNamespace(ns, dynamicClient)
 			}
-		}
+		}()
 	}
 
+	log.Printf("Waiting on wait group")
```

Comment:
Maybe rename to something more user-friendly, like "Dumping Config Connector objects to report.tar.gz"